lib/util/application-logging.js

```diff
@@ -82,10 +82,11 @@ utils.isLogForwardingEnabled = function isLogForwardingEnabled(config, agent) {
  * @param {object} metrics metrics module
  */
 utils.incrementLoggingLinesMetrics = function incrementLoggingLinesMetrics(level, metrics) {
-  const loggingLevelMetric = level ? LOGGING.LEVELS[level.toUpperCase()] : LOGGING.LEVELS.UNKNOWN
+  const loggingLevelMetric = level && LOGGING.LEVELS[level.toUpperCase()]
+  const levelMetric = loggingLevelMetric || LOGGING.LEVELS.UNKNOWN
```

Comment:
I almost started writing a truth table to figure this one out. Since this has already slipped by *twice* as a bug, maybe there's a better way to write it.

Is this clearer?

```js
let levelMetric = LOGGING.LEVELS.UNKNOWN;
if (level) {
  const loggingLevelMetric = LOGGING.LEVELS[level.toUpperCase()]
  if (loggingLevelMetric) {
    levelMetric = loggingLevelMetric
  }
}
```

Reply:
Meh. Not really a fan of nested if statements. This was missed before because the original issue was that level was undefined. It was undefined because we use pinos default levels to map numbers to strings. So a custom level was undefined. Whereas in Winston level is always defined just not one of the ones we support for metric name

Reply:
Okay, but I think the change you have proposed is difficult to read. For instance, I was thinking, "you're telling me that `x ? y : z` is not logically equivalent to `x && y || z`?" and it took me a while to realise why it's not the same.

Reply:
that's fair. I'm ok with changing if it's unclear but your proposal is just a verbose form of it.

Reply:
I agree that we should make this more readable, or maybe abstract into a function, something like `getLogLevel`?